src/node_file.cc

```diff
@@ -2748,6 +2748,11 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
   CHECK_GE(argc, 2);
 
   BufferValue tmpl(isolate, args[0]);
+  const char* ex = "XXXXXX";
+  const auto length = tmpl.length();
+  tmpl.AllocateSufficientStorage(length + strlen(ex));
```

Comment:
I think we also need to call `SetLengthAndZeroTerminate` after allocating the new space.

If we're scraping for even more perf it might be worth doing adding a new `BufferValue` constructor that takes an "additional size" param or something so that we don't end up calling `AllocateSufficientStorage` twice.  There's only two codepaths to pipe that through and it may move the needle a little bit here.

Reply:
It seems it's not required, but, @joyeecheung @jasnell wdyt?